src/VisualStudio/Core/Def/Implementation/InheritanceMargin/InheritanceMarginHelpers.cs

```diff
@@ -135,5 +143,221 @@ public static ImmutableArray<MenuItemViewModel> CreateMenuItemViewModelsForMulti
+        /// </summary>
+        private static TextBlock FormatTaggedText(
+            ClassificationTypeMap classificationTypeMap,
+            IClassificationFormatMap classificationFormatMap,
+            string contentTemplate,
+            params ImmutableArray<TaggedText>[] taggedTexts)
+        {
+            Contract.ThrowIfTrue(taggedTexts.Length == 0);
+            using var _ = CodeAnalysis.PooledObjects.ArrayBuilder<Inline>.GetInstance(out var inlinesBuilder);
+            using var __ = CodeAnalysis.PooledObjects.ArrayBuilder<int>.GetInstance(out var indexBuilder);
+
+            // 1. Find all the indexes of the placeholders from the template.
+            for (var i = 0; i < taggedTexts.Length; i++)
+            {
+                indexBuilder.Add(contentTemplate.IndexOf($"{{{i}}}", StringComparison.Ordinal));
+            }
+
+            var indexArray = indexBuilder.ToImmutable();
+            Contract.ThrowIfTrue(taggedTexts.Length != indexArray.Length);
+            var lengthOfThePlaceholder = "{0}".Length;
+
+            // 2. Spilt the template and insert Inline to the builder
+            for (var i = 0; i < indexArray.Length; i++)
+            {
+                // Have two index point to the start of current placeholder and the end of the previous placeholder
+                // ".....{0}.....{1}....{2}...."
+                //         ↑     ↑
+                //    prevIndex  currentIndex
+                // to insert the string between {0} and {1}
+                var endOfPreviousPlaceHolder = i == 0 ? 0 : indexArray[i - 1] + lengthOfThePlaceholder;
+                var currentIndex = indexArray[i];
+                var prefixString = contentTemplate[endOfPreviousPlaceHolder..currentIndex];
+                inlinesBuilder.Add(new Run(prefixString));
+
+                // Add the TaggedTexts
+                var currentTaggedText = taggedTexts[i];
+                inlinesBuilder.AddRange(currentTaggedText.ToInlines(classificationFormatMap, classificationTypeMap));
+
+                // If this is the last placeholder, try to see if we need to append the suffix string to the end
+                // ".....{0}.....{1}....{2}...."
+                //                      ↑
+                //                      currentIndex
+                if (i == indexArray.Length - 1 && currentIndex + lengthOfThePlaceholder < contentTemplate.Length)
+                {
+                    var suffixString = contentTemplate[(currentIndex + lengthOfThePlaceholder)..];
+                    if (!string.IsNullOrEmpty(suffixString))
+                    {
+                        inlinesBuilder.Add(new Run(suffixString));
+                    }
+                }
+            }
+
+            var toolTipTextBlock = inlinesBuilder.ToTextBlock(classificationFormatMap);
+            toolTipTextBlock.FlowDirection = FlowDirection.LeftToRight;
+            return toolTipTextBlock;
```

Comment:
i have no idea what is going on here.

Reply:
This is just generate the tooltip content (which is just a copy & paste from the old impl). Let me find if this can be merged into the xaml file